src/Reflector/Reflector.php

```diff
@@ -4,18 +4,131 @@
 
 namespace Roave\BetterReflection\Reflector;
 
+use Roave\BetterReflection\Identifier\Identifier;
+use Roave\BetterReflection\Identifier\IdentifierType;
 use Roave\BetterReflection\Reflection\Reflection;
+use Roave\BetterReflection\Reflection\ReflectionClass;
+use Roave\BetterReflection\Reflection\ReflectionConstant;
+use Roave\BetterReflection\Reflection\ReflectionFunction;
 use Roave\BetterReflection\Reflector\Exception\IdentifierNotFound;
+use Roave\BetterReflection\SourceLocator\Type\SourceLocator;
 
-/**
- * This interface is used to ensure a reflector implements these basic methods.
- */
-interface Reflector
+use function assert;
+
+class Reflector
 {
+    public function __construct(private SourceLocator $sourceLocator)
+    {
+    }
+
+    /**
+     * Create a ReflectionClass for the specified $className.
+     *
+     * @return ReflectionClass
+     *
+     * @throws IdentifierNotFound
+     */
+    public function reflectClass(string $identifierName): Reflection
+    {
+        $identifier = new Identifier($identifierName, new IdentifierType(IdentifierType::IDENTIFIER_CLASS));
+
+        $classInfo = $this->sourceLocator->locateIdentifier($this, $identifier);
+        assert($classInfo instanceof ReflectionClass || $classInfo === null);
+
+        if ($classInfo === null) {
+            throw Exception\IdentifierNotFound::fromIdentifier($identifier);
+        }
+
+        return $classInfo;
+    }
+
+    /**
+     * Get all the classes available in the scope specified by the SourceLocator.
+     *
+     * @return ReflectionClass[]
```

Comment:
Since the signature changed, I think it might be a good idea to return an `iterable` here for future-proofing our ability to use a lazy iterator